lib/backend_solver.ml

```diff
@@ -34,14 +34,15 @@ let remote_solve con job request =
   Capnp_rpc_lwt.Capability.with_ref build_job
     (Current_ocluster.Connection.run_job ~job)
 
-let local ?solver_dir () : t =
-  Local
-    (Lwt.return (Solver_worker.spawn_local ?solver_dir ~internal_workers:20 ()))
+let local () : t =
+  Local (Lwt.return (Solver_worker.Solver_request.create ~n_workers:20 ()))
```

Comment:
Note that this is quite a significant change: `spawn_local` created a child process once at startup, which then spawned solver worker processes as needed. Whereas `Solver_request.create` runs the manager in the ocaml-ci process, spawning child processes when the opam-repository commit changes.

The reason for the previous (more complicated) design was to avoid accidentally inheriting file descriptors that either ocaml-ci or the solver forgot to mark as close-on-exec. However, I guess it's been working OK so maybe it's not a problem